src/components/drag-and-drop/custom-drag-layer/custom-drag-layer.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { DragLayer } from 'react-dnd';
+import classNames from 'classnames';
+
+const collect = (monitor) => {
+  const item = monitor.getItem();
+  return {
+    currentOffset: monitor.getSourceClientOffset(),
+    item,
+    draggableNode: item ? item.draggableNode : null,
+    isDragging: monitor.isDragging()
+  };
+};
+
+class CustomDragLayer extends React.Component {
+  static propTypes = {
+    /**
+     * Custom className
+     *
+     * @property className
+     * @type {String}
+     */
+    className: PropTypes.string,
+
+    /**
+     * The dom node being dragged.
+     *
+     * @property draggableNode
+     * @type {Node|Object}
+     */
+    draggableNode: PropTypes.oneOfType([
+      PropTypes.object,
+      PropTypes.node
+    ]),
+
+    /**
+     * Determine if the component is being dragged or not.
+     *
+     * @property isDragging
+     * @type {Boolean}
+     */
+    isDragging: PropTypes.bool.isRequired
+  }
+
+  componentDidMount() {
+    if (this.props.draggableNode) {
+      this.setClonedChildWidth();
+    }
+    if (this.props.draggableNode && this.props.isDragging) {
+      this.createClonedChild();
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props.draggableNode && (prevProps.draggableNode !== this.props.draggableNode)) {
+      this.setClonedChildWidth();
```

Comment:
should we only do this when we need to (when we're dragging or creating the clone?)